src/main/java/io/web/rest/ItemResource.java

```diff
@@ -142,7 +152,12 @@ public ItemResource(ItemRepository itemRepository, UserRepository userRepository
                                                 @RequestParam(value = "category3", required = false) Category category3) {
         log.debug("REST request to get items of logged User");
         log.debug("REST request to get a page of Items");
-        long userId = userRepository.findOneByLogin(SecurityUtils.getCurrentUserLogin().get()).get().getId();
+        Optional<User> optionalUser = userService.getUserWithAuthorities();
+        if(!optionalUser.isPresent()){
+            throw new BadRequestAlertException("there is no logged in user","","");
+        }
+        long userId = optionalUser.get().getId();
```

Comment:
Can be in-lined as
`long userId = userService.getUserWithAuthorities().orElseThrow(() -> new BadRequestAlertException("there is no logged in user","","").getId();`
Also opens a question: why pass `User` if all what is used is `userId` ?